cfme/fixtures/bzs.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+"""Collection of fixtures for simplified work with bzs.
+
+The main purpose of this file is to add a pytest option for generates a BZ report, that
+gives information about the BZs that appear as coverage/automates metadata in the test_functions
+"""
+import pytest
+import yaml
+
+from cfme.fixtures.pytest_store import store
+from cfme.utils.blockers import BZ
+
+
+def pytest_addoption(parser):
+    group = parser.getgroup('Blocker options')
+    group.addoption(
+        '--generate-bz-report',
+        action='store_true',
+        default=False,
+        dest='generate_bz_report',
+        help='Generate a BZ report based on the automates/coverage metadata of test cases.'
+    )
+
+
+@pytest.mark.trylast
+def pytest_collection_modifyitems(session, config, items):
+    if not config.getvalue("generate_bz_report"):
+        return
+    store.terminalreporter.write("Loading automated/covered BZs ...\n", bold=True)
+    bz_list = []
+    for item in items:
```

Comment:
Just general comment about formatting and structure here. Consider using comprehension like below - but note that its kinda depends on whether or not `item._metadata['type']`  always returns a list.

```python
relevant_items = [i for i in items if 'automates' not in i._metadata and 'coverage' not in i._metadata]
bz_list = [i._metadata.get('automates') for i in relevant_items]  # will put some None's in the list
# using extend here depends on the metadata value being a list
[bz_list.extend(i._metadata.get('coverage', [None]) for i in relevant_items]  
bz_list = set([bug.bug_id if isinstance(bug, BZ) else bug for bug in bz_list])  # eliminate duplicates and convert instances
bz_list.remove(None)  # drop the irrelevant None
```

Reply:
Just realized I never responded to this. While this reduces lines of code, in my opinion, what I have there now is a bit more readable. What do you think?